src/com/sahokia/kos/concerttickets/model/Identifiable.java

```diff
@@ -0,0 +1,49 @@
+package com.sahokia.kos.concerttickets.model;
+
+import com.sahokia.kos.concerttickets.annotations.NullableWarning;
+
+import java.lang.reflect.Field;
+
+public abstract class Identifiable {
+
+    @NullableWarning
+    private String id;
+
+    protected Identifiable() {
+        id = null;
+        this.validateFields(this.getClass());
+    }
+
+    public String getId() {
+        return id;
+    }
+
+    public void setId(String id) {
+        if (id == null) {
+            throw new IllegalArgumentException("ID can not be null");
+        }
+        this.id = id;
+    }
+
+    private void validateFields(Class<?> clazz) {
```

Comment:
Validation seems like the other responsibility here -> lets move to the separate class

Reply:
Removed the validation to the separate class